File: sample/me/xiaopan/android/imageloader/sample/MyApplication.java
/*
 * Copyright (C) 2013 Peng fei Pan <sky@xiaopan.me>
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.xiaopan.android.imageloader.sample;

import me.xiaoapn.android.imageloader.R;
import me.xiaopan.android.imageloader.ImageLoader;
import me.xiaopan.android.imageloader.display.ZoomOutBitmapDisplayer;
import me.xiaopan.android.imageloader.process.CircleBitmapProcessor;
import me.xiaopan.android.imageloader.process.ReflectionBitmapProcessor;
import me.xiaopan.android.imageloader.process.RoundedCornerBitmapProcessor;
import me.xiaopan.android.imageloader.task.display.DisplayOptions;
import android.app.Application;

public class MyApplication extends Application {

	@Override
	public void onCreate() {
		super.onCreate();

        ImageLoader.getInstance(getBaseContext()).getConfiguration().setDebugMode(true) //开启Debug，在控制台输出LOG
        .putOptions(DisplayOptionsType.GRID_VIEW, new DisplayOptions(getBaseContext())
            .setLoadingDrawable(R.drawable.image_loading)
            .setLoadFailDrawable(R.drawable.image_load_fail)
            .setProcessor(new ReflectionBitmapProcessor()))
        .putOptions(DisplayOptionsType.VIEW_PAGER, new DisplayOptions(getBaseContext())
            .setLoadFailDrawable(R.drawable.image_load_fail)
            .setDisplayer(new ZoomOutBitmapDisplayer()))
        .putOptions(DisplayOptionsType.LIST_VIEW, new DisplayOptions(getBaseContext())
            .setLoadingDrawable(R.drawable.image_loading)
            .setLoadFailDrawable(R.drawable.image_load_fail)
            .setProcessor(new CircleBitmapProcessor()))
        .putOptions(DisplayOptionsType.GALLERY, new DisplayOptions(getBaseContext())
            .setLoadingDrawable(R.drawable.image_loading)
            .setLoadFailDrawable(R.drawable.image_load_fail)
            .setProcessor(new RoundedCornerBitmapProcessor()));
	}
}

File: samples/androidApp/src/main/kotlin/com/github/panpf/sketch/sample/MyApplication.kt
/*
 * Copyright (C) 2024 panpf <panpfpanpf@outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.panpf.sketch.sample

import android.annotation.SuppressLint
import android.app.Application
import com.github.panpf.sketch.sample.ui.gallery.PhotoPaletteViewModel
import com.github.panpf.sketch.sample.ui.setting.AppSettingsViewModel
import com.github.panpf.sketch.sample.ui.test.DrawableScaleTypeViewModel
import com.github.panpf.sketch.sample.ui.test.LocalVideoListViewModel
import com.github.panpf.sketch.sample.ui.test.transform.BlurTransformationTestViewModel
import com.github.panpf.sketch.sample.ui.test.transform.CircleCropTransformationTestViewModel
import com.github.panpf.sketch.sample.ui.test.transform.MaskTransformationTestViewModel
import com.github.panpf.sketch.sample.ui.test.transform.MultiTransformationTestViewModel
import com.github.panpf.sketch.sample.ui.test.transform.RotateTransformationTestViewModel
import com.github.panpf.sketch.sample.ui.test.transform.RoundedCornersTransformationTestViewModel
import org.koin.core.module.Module
import org.koin.core.module.dsl.viewModel
import org.koin.core.module.dsl.viewModelOf
import org.koin.dsl.module
import java.security.SecureRandom
import java.security.cert.X509Certificate
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLContext
import javax.net.ssl.TrustManager
import javax.net.ssl.X509TrustManager

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        handleSSLHandshake()
        initialApp(this@MyApplication) {
            modules(viewModule())
        }
    }

    private fun viewModule(): Module = module {
        viewModel { AppSettingsViewModel(sketch = get(), appSettings = get(), page = it.get()) }
        viewModelOf(::LocalVideoListViewModel)
        viewModelOf(::DrawableScaleTypeViewModel)
        viewModelOf(::BlurTransformationTestViewModel)
        viewModelOf(::MaskTransformationTestViewModel)
        viewModelOf(::MultiTransformationTestViewModel)
        viewModelOf(::RotateTransformationTestViewModel)
        viewModelOf(::CircleCropTransformationTestViewModel)
        viewModelOf(::RoundedCornersTransformationTestViewModel)
        viewModelOf(::PhotoPaletteViewModel)
    }

    // TODO apigiphy.com is not accessible on Android
    //  The simulator reports SSLHandshakeException: connection closed
    //  A real machine must be bypassed to gain access.
    /**
     * for api.pexels.com on Android 5.0
     */
    private fun handleSSLHandshake() {
        try {
            val trustAllCerts = arrayOf<TrustManager>(
                @SuppressLint("CustomX509TrustManager")
                object : X509TrustManager {
                    override fun getAcceptedIssuers(): Array<X509Certificate?> {
                        return arrayOfNulls(0)
                    }

                    @SuppressLint("TrustAllX509TrustManager")
                    override fun checkClientTrusted(
                        certs: Array<X509Certificate?>?,
                        authType: String?
                    ) {
                    }

                    @SuppressLint("TrustAllX509TrustManager")
                    override fun checkServerTrusted(
                        certs: Array<X509Certificate?>?,
                        authType: String?
                    ) {
                    }
                })
            val sc = SSLContext.getInstance("TLS")
            // trustAllCerts trust all certificates
            sc.init(null, trustAllCerts, SecureRandom())
            HttpsURLConnection.setDefaultSSLSocketFactory(sc.socketFactory)
            HttpsURLConnection.setDefaultHostnameVerifier { _, _ -> true }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}